app/src/main/java/org/fossasia/openevent/general/favorite/FavouriteEventsViewModel.kt

```diff
@@ -54,4 +62,53 @@ class FavouriteEventsViewModel(private val eventService: EventService) : ViewMod
         super.onCleared()
         compositeDisposable.clear()
     }
+
+    fun searchToday(calendar: Calendar, searchTimeViewModel: SearchTimeViewModel, activity: AppCompatActivity?) {
+        searchTimeViewModel.saveDate(EventUtils.getSimpleFormattedDate(calendar.time))
+        calendar.add(Calendar.DATE, 1)
+        searchTimeViewModel.saveNextDate(EventUtils.getSimpleFormattedDate(calendar.time))
+        searchTimeViewModel.saveDate(TODAY)
+        redirectToSearch(activity)
+    }
+
+    fun searchTomorrow(calendar: Calendar, searchTimeViewModel: SearchTimeViewModel, activity: AppCompatActivity?) {
+        calendar.add(Calendar.DATE, 1)
+        searchTimeViewModel.saveNextDate(EventUtils.getSimpleFormattedDate(calendar.time))
+        calendar.add(Calendar.DATE, 1)
+        searchTimeViewModel.saveNextToNextDate(EventUtils.getSimpleFormattedDate(calendar.time))
+        searchTimeViewModel.saveDate(TOMORROW)
+        redirectToSearch(activity)
+    }
+
+    fun searchWeekend(calendar: Calendar, searchTimeViewModel: SearchTimeViewModel, activity: AppCompatActivity?) {
+        val today = calendar.get(Calendar.DAY_OF_WEEK)
+        if (today != Calendar.SATURDAY) {
+            val offset = Calendar.SATURDAY - today
+            calendar.add(Calendar.DATE, offset)
+        }
+        searchTimeViewModel.saveWeekendDate(EventUtils.getSimpleFormattedDate(calendar.time))
+        calendar.add(Calendar.DATE, 1)
+        searchTimeViewModel.saveNextToWeekendDate(EventUtils.getSimpleFormattedDate(calendar.time))
+        searchTimeViewModel.saveDate(THIS_WEEKEND)
+        redirectToSearch(activity)
+    }
+
+    fun searchNextMonth(calendar: Calendar, searchTimeViewModel: SearchTimeViewModel, activity: AppCompatActivity?) {
```

Comment:
These four methods can combine and parameters can pass accordingly. @iamareebjamal Please make me correct.